cmd/digestGenome/digestGenome.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fasta"
+	"github.com/vertgenlab/gonomics/fileio"
+	"log"
+	"strings"
+)
+
+type restrictionEnzyme struct {
+	name       string
+	cutString  string
+	cutBases   []dna.Base
+	cutBasesRC []dna.Base
+	cutPos     int
+	pal        bool //palindrome
+}
+
+func getCutPos(cutSite string) (cutPos int, cutString []string) {
+	var found bool = false
+	for i := range cutSite {
+		if string(cutSite[i]) == "^" {
+			found = true
+			cutPos = i
+		} else {
+			cutString = append(cutString, string(cutSite[i]))
+		}
+	}
+	if found == false {
+		log.Fatalf("The input restriction enzyme cut site must have the '^' character to denote the cut location. Your seq: %s", cutSite)
+	}
+	return cutPos, cutString
+}
+
+func digestGenome(genome string, cutSite string, outFile string) {
+	var currChrom string
+	var revMatch bool = false
+	var cutString []string
+	var base, prevCut, numCut int
+	var bedRegion bed.Bed
+	var re restrictionEnzyme
+
+	out := fileio.EasyCreate(outFile)
+
+	//default enzymes
+	switch cutSite {
+	case "MboI":
+		re.name = cutSite
+		re.cutPos = 0
+		re.cutBases = []dna.Base{2, 0, 3, 1}
+		re.cutBasesRC = dna.ReverseComplementAndCopy(re.cutBases)
+		re.pal = true
+	case "DnpII":
+		re.name = cutSite
+		re.cutPos = 0
+		re.cutBases = []dna.Base{2, 0, 3, 1}
+		re.cutBasesRC = dna.ReverseComplementAndCopy(re.cutBases)
+		re.pal = true
+	case "BglII":
+		re.name = cutSite
+		re.cutPos = 1
+		re.cutBases = []dna.Base{0, 2, 0, 3, 1, 3}
+		re.cutBasesRC = dna.ReverseComplementAndCopy(re.cutBases)
+		re.pal = true
+	case "HindIII":
+		re.name = cutSite
+		re.cutPos = 1
+		re.cutBases = []dna.Base{0, 0, 2, 1, 3, 3}
+		re.cutBasesRC = dna.ReverseComplementAndCopy(re.cutBases)
+		re.pal = true
+	default:
+		re.name = cutSite
+		re.cutPos, cutString = getCutPos(cutSite)
+		re.cutBases = dna.StringToBases(strings.Join(cutString, ""))
+		re.cutBasesRC = dna.ReverseComplementAndCopy(re.cutBases)
+		if dna.CompareSeqsIgnoreCase(re.cutBases, re.cutBasesRC) == 0 {
+			re.pal = true
+		} else {
+			re.pal = false
+		}
+	}
+
+	fa := fasta.GoReadToChan(genome)
+	for i := range fa {
+		prevCut = 0 //set the first fragment start base to 0
+		numCut = 0
+		currChrom = i.Name
+		for base = 0; base <= len(i.Seq)-len(re.cutBases); base++ { //loop over every base in the fasta file
+			testBases := i.Seq[base : base+len(re.cutBases)] //get the next base in the fasta file plus the number of bases in the recognition motif. I don't get an index out of range error at the end of the sequence--not sure why
```

Comment:
I would declare the testBases variable outside the loop to cut down on allocations

Reply:
good catch